src/Lean/Environment.lean

```diff
@@ -1755,6 +1769,18 @@ def mkModuleData (env : Environment) (level : OLeanLevel := .private) : IO Modul
     constNames, constants, entries
   }
 
+@[extern "lean_ir_export_entries"]
+private opaque exportIREntries (env : Environment) : Array (Name × Array EnvExtensionEntry)
+
+private def mkIRData (env : Environment) : IO ModuleData := do
+  -- TODO: should we use a more specific/efficient data format for IR?
```

Comment:
I guess not really relevant for landing this, but what are you thinking of regarding increased efficiency?

Reply:
Some bytecode more or less; anything without indirections even for straight-line code :)